cmd/access/node_builder/access_node_builder.go

```diff
@@ -590,14 +626,120 @@ func (builder *FlowAccessNodeBuilder) BuildExecutionSyncComponents() *FlowAccess
 				builder.State,
 				builder.Storage.Headers,
 				builder.executionDataConfig,
+				execDataDistributor,
 			)
 
 			builder.FollowerDistributor.AddOnBlockFinalizedConsumer(builder.ExecutionDataRequester.OnBlockFinalized)
-			builder.ExecutionDataRequester.AddOnExecutionDataReceivedConsumer(execDataDistributor.OnExecutionDataReceived)
+
+			// add requester into ReadyDoneAware dependency passed to indexer. This allows the indexer
+			// to wait for the requester to be ready before starting.
+			requesterDependable.Init(builder.ExecutionDataRequester)
 
 			return builder.ExecutionDataRequester, nil
 		})
 
+	if builder.executionDataIndexingEnabled {
```

Comment:
I see there are a couple of scenarios where building a component depends on a boolean flag, would it make sense to have a builder option for it, something like so:
```go
builder.
    Requires(builder.executionDataIndexingEnabled).
    Module("...") // ...
```

If the value is false the whole procedure would stop.

Reply:
yea, that's a cool idea. It would require a bit of refactoring in the node builder, but could make some of the bootstrapping logic cleaner.

There aren't a ton of places we load conditional modules/components like this though. mostly in the AN/ON codebase since they have more user options